### PR TITLE
Remove deprecated Shutdown Room and Purge Room Admin API

### PR DESCRIPTION
**Pull request by dklimpel**
Follow up #8829 
Fixes: #8663, #9052 
Related: #8810 

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

Signed-off-by: Dirk Klimpel dirk@klimpel.org

### CODE REVIEW

**Comment by turt2live** [comment]
fyi these endpoints are still used in tooling which is in active circulation - I don't think it's a good idea to remove them at the moment, particularly considering it's never been clear that they are deprecated.

**Comment by dklimpel** [comment]
Yes, you're right. It is only the preparation. It was one workflow to create both PR. Deprecation and removal. I know that this PR needs a few month till merge.

**Comment by clokep** [comment]
This was removed in v1.25.0, I wonder if it is time to revisit this?

**Comment by dklimpel** [comment]
> This was removed in v1.25.0, I wonder if it is time to revisit this?

I am waiting for https://github.com/matrix-org/mjolnir/issues/76

**Comment by dklimpel** [comment]
I would like to finish this.
See also: https://github.com/matrix-org/synapse/issues/9052#issuecomment-853790314
> We'd like to do this by August, will discuss with Mjolnir owners

**Comment by richvdh** [comment]
fixes: #5575

**Comment by dklimpel** [comment]
The current PR we are waiting for: https://github.com/matrix-org/mjolnir/pull/117

File: docs/upgrade.md [hunk 1]
```diff
@@ -85,6 +85,20 @@ process, for example:
...
+
+Remove old Room Admin API:
+
+-   `POST /_synapse/admin/v1/purge_room`
+-   `POST /_synapse/admin/v1/shutdown_room/<room_id>`
+
+The new [Delete Room Admin API](https://matrix-org.github.io/synapse/develop/admin_api/rooms.html#delete-room-api)
+is accessible under `POST /_synapse/admin/v1/rooms/<room_id>/delete`.
+
+The deprecation of the old endpoints was announced with Synapse 1.25.0 (released on 2021-01-13).
```
Comment (richvdh) [inline docs/upgrade.md:100]: Some minor wording tweaks here

```suggestion
## Removal of old Room Admin API

The following admin APIs were deprecated in [Synapse 1.25](https://github.com/matrix-org/synapse/blob/v1.25.0/CHANGES.md#removal-warning)
(released on 2021-01-13) and have now been removed:

-   `POST /_synapse/admin/v1/purge_room`
-   `POST /_synapse/admin/v1/shutdown_room/<room_id>`

Any scripts still using the above APIs should be converted to use the 
[Delete Room API](https://matrix-org.github.io/synapse/latest/admin_api/rooms.html#delete-room-api).
```

**Review by richvdh — APPROVED** [review]
lgtm otherwise, but we're going to wait for https://github.com/matrix-org/mjolnir/pull/117 to land.

File: synapse/rest/admin/rooms.py [hunk 2]
```diff
@@ -46,6 +46,48 @@
 logger = logging.getLogger(__name__)
 
 
+class ResolveRoomIdMixin:
```
Comment (richvdh) [inline synapse/rest/admin/rooms.py:49]: where did this come from ? :thinking:

**Comment by richvdh** [comment]
ugh, I think I did the merge wrong :/

**Comment by dklimpel** [comment]
> ugh, I think I did the merge wrong :/

It was confusing. I hope it is better now.

**Review by richvdh — APPROVED** [review]
thank you for fixing my mess, @dklimpel!

**Pull request merged**